src/NATS.Client.JetStream/Internal/ReplyToDateTimeAndSeq.cs

```diff
@@ -0,0 +1,28 @@
+namespace NATS.Client.JetStream.Internal;
+
+internal static class ReplyToDateTimeAndSeq
+{
+    internal static (DateTimeOffset? DateTime, long? Seq) Parse(string? reply)
+    {
+        if (reply == null)
+        {
+            return (null, null);
+        }
+
+        var ackSeperated = reply.Split(".");
+
+        // It must be seperated by 9 dots as defined in
+        // https://docs.nats.io/reference/reference-protocols/nats_api_reference#acknowledging-messages
+        if (ackSeperated.Length != 9)
```

Comment:
This has couple of variations actually. See the design doc:
https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-15.md#jsack

you can also check the Go client implementation if you like:
https://github.com/nats-io/nats.go/blob/main/internal/parser/parse.go#L57

Reply:
Oh I didn't know that there is also a v2. Funny that the official docs says nothing about it.